internal/controller/nopresource/nopresource.go

```diff
@@ -108,57 +108,29 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
 		return managed.ExternalObservation{ResourceExists: false}, nil
 	}
 
-	// Logic 1 - Set condition as soon as the time elapsed in seconds is equal to time specified in Spec
-	// for i := 0; i < len(cr.Spec.ForProvider.ConditionAfter); i++ {
-	// 	elapsed, _ := time.ParseDuration(cr.Spec.ForProvider.ConditionAfter[i].Time)
-	// 	if time.Duration(time.Since(startTime.Time).Seconds()) == time.Duration(elapsed.Seconds()) {
-	// 		fmt.Printf("Calling update on index %d\n", i)
+	ci := ReconcileLogic(cr.Spec.ForProvider.ConditionAfter, time.Since(startTime.Time))
 
-	// 		x := runtimev1alpha1.Condition{
-	// 			Type:               runtimev1alpha1.ConditionType(cr.Spec.ForProvider.ConditionAfter[i].ConditionType),
-	// 			Status:             v1.ConditionStatus(cr.Spec.ForProvider.ConditionAfter[i].ConditionStatus),
-	// 			LastTransitionTime: metav1.Now(),
-	// 			Reason:             runtimev1alpha1.ReasonAvailable,
-	// 		}
+	for _, l := range ci {
+		// fmt.Printf("Calling update on index %d\n", l)
 
-	// 		cr.Status.SetConditions(x)
-
-	// 	}
-	// }
-
-	// Logic 2 - Set condition at every reconcile to the latest specified condition in Spec
-	l := -1
-	mx := -1
-	for i := 0; i < len(cr.Spec.ForProvider.ConditionAfter); i++ {
-		elapsed, _ := time.ParseDuration(cr.Spec.ForProvider.ConditionAfter[i].Time)
-		if time.Duration(time.Since(startTime.Time).Seconds()) >= time.Duration(elapsed.Seconds()) {
-			if mx < int(elapsed.Seconds()) {
-				mx = int(elapsed.Seconds())
-				l = i
-			}
-		}
-	}
-	if l != -1 {
-		fmt.Printf("Calling update on index %d\n", l)
-
-		x := runtimev1alpha1.Condition{
-			Type:               runtimev1alpha1.ConditionType(cr.Spec.ForProvider.ConditionAfter[l].ConditionType),
+		x := xpv1.Condition{
+			Type:               xpv1.ConditionType(cr.Spec.ForProvider.ConditionAfter[l].ConditionType),
 			Status:             v1.ConditionStatus(cr.Spec.ForProvider.ConditionAfter[l].ConditionStatus),
 			LastTransitionTime: metav1.Now(),
-			Reason:             runtimev1alpha1.ReasonAvailable,
+			Reason:             xpv1.ReasonAvailable,
```

Comment:
this Reason is not always going to be accurate depending on what the condition actually is - Ready (true/false) and Synced (true/false) all have different reasons associated with them: https://github.com/crossplane/crossplane-runtime/blob/master/apis/common/v1/condition.go#L43-L54